### PR TITLE
Move bug

### PR DESCRIPTION
**Pull request by McCls**
### Type of Change (New Feature, Bugfix, Refactor, etc)
Bugfix

### Summary of Change
Changing the way in which moves are preformed. Order of operations is now:
1. Validate that the requested move is allowed (if it is not allowed, stop here and do nothing)
2. Play move out animation
3. Actually perform requested move
4. Play move in animation

### Justification of Change
Fixing a bug

### Link to covered ticket(s)
#17 

### Tested? [ x]
Tested in game

### CODE REVIEW

File: Assets/Code/Sprite/AnimatedSprite/Character/Character.cs [hunk 1]
```diff
@@ -43,13 +43,16 @@ IEnumerator Idle()
...
+    public delegate bool MoveFunction(Character character);
+
+    IEnumerator MoveOut(MoveFunction moveFunction)
     {
         for(int i = 0; i < move.Length; i++)
         {
             spriteRenderer.sprite = move[i];
             yield return new WaitForSeconds(moveAnimationTime);
         }
+        moveFunction(this);
```
Comment (McCls) [inline Assets/Code/Sprite/AnimatedSprite/Character/Character.cs:58]:
```suggestion

        moveFunction(this);
```

File: Assets/Code/Sprite/PositionGrid/PositionGrid.cs [hunk 2]
```diff
@@ -89,6 +89,31 @@ private bool Move(Character character, GridCoordinates target)
...
+        target.X -= 1;
+        return BoundCheck(target.X, target.Y);
+    }
+    public bool MoveRightValid(Character character)
+    {
+        GridCoordinates target = character.Position;
+        target.X += 1;
+        return BoundCheck(target.X, target.Y);
+    }
+
```
Comment (darin-ellis7) [inline Assets/Code/Sprite/PositionGrid/PositionGrid.cs]: I don't like the idea of duplicating the move validation code from the actual move functions to these new ones. There's probably a better way to structure this entirely. Perhaps have these validity functions inside the functions that will have the  actual movement? They are already returning the BoundCheck() boolean.

File: Assets/Code/Sprite/AnimatedSprite/Character/Player/Player.cs [hunk 3]
```diff
@@ -9,30 +9,30 @@ void Update()
...
+                PlayMovementAnimation(grid.MoveDown);
             }
         }
         if(Input.GetKeyDown("d"))
         {
-            if(grid.MoveRight(this))
+            if(grid.MoveRightValid(this))
             {
-                PlayMovementAnimation();
+                PlayMovementAnimation(grid.MoveRight);
```
Comment (darin-ellis7) [inline Assets/Code/Sprite/AnimatedSprite/Character/Player/Player.cs:35]: Maybe PositionGrid should be the one to kick off the animation? That may be a more elegant solution: BoundCheck -> Start animation, wait as necessary -> Execute actual move.

**Review by darin-ellis7 — APPROVED** [review]
LGTM. The general structure is good. As decided over voicechat though, the specific implementation speaks to the need for a separate Move class to mediate between the Character and the PositionGrid. This will be spun out and handled by https://github.com/darin-ellis7/GridGame/issues/21

**Pull request merged**